packages/ember/tests/routing/router_service_test.js

```diff
@@ -0,0 +1,432 @@
+        assert.equal(routerService.get('currentURL'), '/');
+      });
+    }
+
+    ['@test RouterService#currentURL is correctly set for child route'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/child').then(() => {
+        assert.equal(routerService.get('currentURL'), '/child');
+      });
+    }
+
+    ['@test RouterService#currentURL is correctly set after transition'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        },
+
+        afterModel() {
+          this.transitionTo('parent.sister');
+        }
+      }));
+
+      return this.visit('/child').then(() => {
+        assert.equal(routerService.get('currentURL'), '/sister');
+      });
+    }
+
+    ['@test RouterService#currentURL is correctly set on each transition'](assert) {
+      assert.expect(3);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/child')
+        .then(() => {
+          assert.equal(routerService.get('currentURL'), '/child');
+
+          return this.visit('/sister');
+        })
+        .then(() => {
+          assert.equal(routerService.get('currentURL'), '/sister');
+
+          return this.visit('/brother');
+        })
+        .then(() => {
+            assert.equal(routerService.get('currentURL'), '/brother');
+        });
+    }
+
+    ['@test RouterService#rootURL is correctly set to the default value'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.index', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/').then(() => {
+        assert.equal(routerService.get('rootURL'), '/');
+      });
+    }
+
+    ['@test RouterService#rootURL is correctly set to a custom value'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.index', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          set(this.router, 'rootURL', '/test');
```

Comment:
I have a slight preference to use something other than `/test` here (since when tests are running we are already in `/tests` path).  It definitely doesn't matter for this test as written, but it would just make this a tad bit easier for my brain to parse 😝